### PR TITLE
Add ADC driver for nRF51

### PR DESCRIPTION
**Pull request by doceme**

### CODE REVIEW

**Comment by doceme** [comment]
I just realized that I need to fix the copyright in the main.c file.  I'll push an update once the other changes have been reviewed. No rush, btw.

File: os/hal/ports/NRF51/NRF51822/adc_lld.c [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+ *
+ * @notapi
+ */
+void adc_lld_stop(ADCDriver *adcp) {
+
+  /* If in ready state then disables the ADC clock and analog part.*/
+  if (adcp->state == ADC_READY) {
+
+#if NRF51_ADC_USE_ADC1
+    if (&ADCD1 == adcp)
```
Comment (utzig) [inline os/hal/ports/NRF51/NRF51822/adc_lld.c]: Either add braces here of fix identation.

**Comment by utzig** [comment]
You can add commit with fixes to this pull request. Just update the license as you wish and fix the bracing/style in lld_stop. No other issues found on review.

**Comment by doceme** [comment]
Will do.

~Stephen

**Comment by utzig** [comment]
Sorry it took so long to merge this. I was waiting for a notification of your update but Github never showed it (may be my configuration)...

**Comment by doceme** [comment]
No problem.  I just didn't want to rush you.

**Pull request merged**